cmd/net.go

```diff
@@ -378,3 +380,59 @@ func CheckLocalServerAddr(serverAddr string) error {
 
 	return nil
 }
+
+// isDomainName checks if a string is a presentation-format domain name
+// (currently restricted to hostname-compatible "preferred name" LDH labels and
+// SRV-like "underscore labels"; see golang.org/issue/12421).
+func isDomainName(s string) bool {
```

Comment:
can we use this? https://golang.org/src/vendor/golang_org/x/net/lex/httplex/httplex.go?s=4764:4799#L204

Reply:
Actually its better to use https://godoc.org/github.com/miekg/dns#IsDomainName we already depend on this.

Reply:
@harshavardhana fixed, it is using miekg/dns now.